### PR TITLE
Adds etags for web static cached results

### PR DESCRIPTION
**Pull request by wcharczuk**
## PR Summary

Adds etag functionality to the static file server in cached mode.

 - **Type:** Improvements
 - **Intended Change Level:** minor

#### Reviewers:

Reviewers keep the following in mind while reviewing this PR, and provide an assessment of them in your approval comment:

- Does the "Intended Change Level" above match the actual behavior of this PR?
- Is this PR following patterns set forth in the package (if modifying a package), or the go-sdk design patterns if implementing a new package from scratch?
- Does this require a backport to LTS versions? If so ping, let the contributors group know.

### CODE REVIEW

File: web/static_file_server.go [hunk 1]
```diff
@@ -127,6 +130,13 @@ func (sc *StaticFileServer) ServeFile(r *Ctx, filePath string) Result {
 
 	finfo, err := f.Stat()
 	if err != nil {
+		if os.IsNotExist(err) {
```
Comment (dhermes) [inline web/static_file_server.go]: 1. This change seems unrelated to the ETag change?
1. These lines are identical to 155-167, thoughts on factoring out?

File: webutil/etag.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+	"github.com/blend/go-sdk/ex"
+)
+
+// ETag creates an etag for a given blob.
+func ETag(contents []byte) (string, error) {
+	hash := md5.New()
+	_, err := hash.Write(contents)
+	if err != nil {
+		return "", ex.New(err)
```
Comment (dhermes) [inline webutil/etag.go]: Seems quite challenging to cook up a `contents` that will trigger this branch?

Comment (mat285) [reply webutil/etag.go]: This actually never will return an error according to the docs so feel free to not check it https://golang.org/pkg/hash/

Comment (mat285) [reply webutil/etag.go]: It only has it in the spec because of `io.Writer`

File: webutil/etag.go [hunk 3]
```diff
@@ -3,16 +3,11 @@ package webutil
...
 // ETag creates an etag for a given blob.
-func ETag(contents []byte) (string, error) {
+func ETag(contents []byte) string {
 	hash := md5.New()
-	_, err := hash.Write(contents)
-	if err != nil {
-		return "", ex.New(err)
-	}
-	return hex.EncodeToString(hash.Sum(nil)), nil
+	hash.Write(contents)
```
Comment (dhermes) [inline webutil/etag.go]: ISTM `_, _ = hash.Write(contents)` is the better move here, with a comment explaining why it's safe to ignore the error.

If we ever decide to integrate [`errcheck`][1] into CI (which I highly recommend) this will get flagged.

[1]: https://github.com/kisielk/errcheck
